src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp

```diff
@@ -628,6 +699,344 @@ void Renderer_REC2100_Surround::apply(const void * inImg, void * outImg, long nu
+        const float y = in[1];
+        const float Y = in[2];
+
+        const float d = (y == 0.f) ? 0.f : 1.f / y;
+        const float X = Y * x * d;
+        const float Z = Y * (1.f - x - y) * d;
+
+        out[0] = X;
+        out[1] = Y;
+        out[2] = Z;
+        out[3] = in[3];
+
+        in  += 4;
+        out += 4;
+    }
+}
+
+Renderer_XYZ_TO_uvY::Renderer_XYZ_TO_uvY(ConstFixedFunctionOpDataRcPtr & /*data*/)
+    :   OpCPU()
+{   
+}
+
+void Renderer_XYZ_TO_uvY::apply(const void * inImg, void * outImg, long numPixels) const
+{
+    const float * in = (const float *)inImg;
+    float * out = (float *)outImg;
+
+    for(long idx=0; idx<numPixels; ++idx)
+    {
+        // TODO: Check robustness for arbitrary float inputs.
+
+        const float X = in[0];
+        const float Y = in[1];
+        const float Z = in[2];
+
+        float d = X + 15.f * Y + 3.f * Z;
+        d = (d == 0.f) ? 0.f : 1.f / d;
+        const float u = 4.f * X * d;
+        const float v = 9.f * Y * d;
+
+        out[0] = u;
+        out[1] = v;
+        out[2] = Y;
+        out[3] = in[3];
+
+        in  += 4;
+        out += 4;
+    }
+}
+
+Renderer_uvY_TO_XYZ::Renderer_uvY_TO_XYZ(ConstFixedFunctionOpDataRcPtr & /*data*/)
+    :   OpCPU()
+{   
+}
+
+void Renderer_uvY_TO_XYZ::apply(const void * inImg, void * outImg, long numPixels) const
+{
+    const float * in = (const float *)inImg;
+    float * out = (float *)outImg;
+
+    for(long idx=0; idx<numPixels; ++idx)
+    {
+        // TODO: Check robustness for arbitrary float inputs.
+
+        const float u = in[0];
+        const float v = in[1];
+        const float Y = in[2];
+
+        const float d = (v == 0.f) ? 0.f : 1.f / v;
+        const float X = (9.f / 4.f) * Y * u * d;
+        const float Z = (3.f / 4.f) * Y * (4.f - u - 6.666666666666667f * v) * d;
+
+        out[0] = X;
+        out[1] = Y;
+        out[2] = Z;
+        out[3] = in[3];
+
+        in  += 4;
+        out += 4;
+    }
+}
+
+Renderer_XYZ_TO_LUV::Renderer_XYZ_TO_LUV(ConstFixedFunctionOpDataRcPtr & /*data*/)
+    :   OpCPU()
+{   
+}
+
+void Renderer_XYZ_TO_LUV::apply(const void * inImg, void * outImg, long numPixels) const
+{
+    const float * in = (const float *)inImg;
+    float * out = (float *)outImg;
+
+    for(long idx=0; idx<numPixels; ++idx)
+    {
+        // TODO: Check robustness for arbitrary float inputs.
+
+        const float X = in[0];
+        const float Y = in[1];
+        const float Z = in[2];
+
+        float d = X + 15.f * Y + 3.f * Z;
+        d = (d == 0.f) ? 0.f : 1.f / d;
+        const float u = 4.f * X * d;
+        const float v = 9.f * Y * d;
+
+        const float Lstar = ( Y <= 0.008856451679f ) ? 9.0329629629629608f * Y : 
+                                                     1.16f * powf( Y, 0.333333333f ) - 0.16f;
+        const float ustar = 13.f * Lstar * (u - 0.19783001f);   // D65 white
+        const float vstar = 13.f * Lstar * (v - 0.46831999f);   // D65 white
```

Comment:
D65  looks like it should be a parameter at some point. Also I think this is kind of a very basic colour appearance model, which I could see being extended to include the other CAMs Lab, CIECOM02, etc etc. At which point the parameter would be common to these?

Reply:
Thanks for the comment Kevin.  We want to keep it simple for now but there should be no problem extending it in the future to take a parameter as long as the default remains D65.